usage/usage.go

```diff
@@ -0,0 +1,136 @@
+// Package usage implements usage tracking for k6 in order to figure what is being used within a given execution
+package usage
+
+import (
+	"strings"
+	"sync"
+)
+
+// Usage is a way to collect usage data for within k6
+type Usage struct {
+	l *sync.Mutex
+	m map[string]any
+}
+
+// New returns a new empty Usage ready to be used
+func New() *Usage {
+	return &Usage{
+		l: new(sync.Mutex),
+		m: make(map[string]any),
+	}
+}
+
+// String appends the provided value to a slice of strings that is the value.
+// If called only a single time, the value will be just a string not a slice
```

Comment:
If we drop this part - a lot of the code becomes a lot nicer as we only have 2 types.

Also in theory we can go the more typesafe way and actually have separate counters and strings which might help as well ... it will just mean they need to be merged at a later point.

I specifically stopped trying to polish this part as I did a couple of tries and all of them had problems and decided on going with reproducing the current report as well as possible.